packages/foundry/src/bin.ts

```diff
@@ -1,22 +1,46 @@
-import * as process from "process";
+#!/usr/bin/env node
+import path from "node:path";
+import process from "node:process";
+import { Command } from "commander";
+import { type OptionValues } from "@commander-js/extra-typings";
 
-import { Configuration, runCli } from "./cli";
-import { loadTemplate } from "./loadTemplate";
-import { generator } from "./generator";
+import { Templates, create } from "./create.js";
 
-const main = async (): Promise<void> => {
-    const config: Configuration = runCli();
+import packageJson from "../package.json" assert { type: "json" };
 
-    await loadTemplate(config);
+const program = new Command();
 
-    await generator(config.outputDirectory);
-};
+program.name(packageJson.name).description(packageJson.description).version(packageJson.version);
 
-main()
-    .then(() => {
-        process.exit(0);
-    })
-    .catch(error => {
-        console.error(error);
-        process.exit(1);
+for (const [ templateId, template ] of Templates) {
```

Comment:
There is still a strong association between commands and template here. we need to split this a bit more

Reply:
I wrote you a message on Slack